libraries/MySensors/MyHwATMega328.cpp

```diff
@@ -17,6 +17,8 @@
  * version 2 as published by the Free Software Foundation.
  */
 
+#ifdef __AVR_ATmega328P__
```

Comment:
A number of standard #defines are appended to the compiler depending on the target board (you can see it in the commandline behind the e.g. avr-g++ command):

* ESP: ARDUINO_ESP8266_ESP01   ARDUINO_ARCH_ESP8266   ESP8266
* UNO: ARDUINO_AVR_UNO   ARDUINO_ARCH_AVR
* MEGA: ARDUINO_AVR_MEGA2560   ARDUINO_ARCH_AVR
* DUE: ARDUINO_SAM_DUE   ARDUINO_ARCH_SAM  __SAM3X8E__

Then, there are a number or implicit defines introduced by the compiler depending on the target mcu. See e.g. https://gcc.gnu.org/onlinedocs/gcc/AVR-Options.html

I think, to separate between architectures (AVR/ESP/SAM) we can do with the ARDUINO_ARCH_xxx defines.
For this case (and the next one) the #ifdef should be changed to:
#ifdef ARDUINO_ARCH_AVR

**BUT** I find it strange (and that's how this #ifdef sneaked in) that a file called 'MyHw**ATMega328**.cpp' is required by a Mega2560 board.
If this file is indeed shared by all AVR based MCU's, then it should better be called 'MyHwAvr.cpp' or something like it.